src/components/form/AdditionalCustomPackages.js

```diff
@@ -0,0 +1,485 @@
+  <Text className="pf-u-text-align-center pf-u-pr-xl pf-u-pl-xl pf-u-pt-xl">
+    {text}
+  </Text>
+);
+
+EmptyText.propTypes = {
+  text: PropTypes.string,
+};
+
+const NoResultsText = ({ heading, body }) => (
+  <TextContent className="pf-u-text-align-center pf-u-pr-xl pf-u-pl-xl pf-u-pt-xl">
+    <Text component="h3">{heading}</Text>
+    <Text component="small">{body}</Text>
+  </TextContent>
+);
+
+NoResultsText.propTypes = {
+  heading: PropTypes.string,
+  body: PropTypes.string,
+};
+
+const mapPackagesToOptions = (pkgs) =>
+  pkgs.map((pkg) => ({
+    selected: false,
+    isVisible: true,
+    name: pkg.package_name || pkg.name,
+    summary: pkg.summary,
+  }));
+
+const AdditionalCustomPackages = ({ defaultArch, ...props }) => {
+  const { change, getState } = useFormApi();
+  const { input } = useFieldApi(props);
+  const [availableOptions, setAvailableOptions] = React.useState([]);
+  const [chosenOptions, setChosenOptions] = React.useState([]);
+  const [availableInputValue, setAvailableInputValue] = React.useState('');
+  const [enterPressed, setEnterPressed] = useState(false);
+  const [hasMoreResults, setHasMoreResults] = useState(false);
+  const [scrollTo, setScrollTo] = useState(null);
+  const [hasNoSearchResults, setHasNoSearchResults] = useState(false);
+  const [searchFocused, setSearchFocused] = useState(false);
+  const [exactMatch, setExactMatch] = useState(false);
+  const [hasOnlyExactMatch, setOnlyExactMatch] = useState(false);
+
+  useEffect(() => {
+    const loadedPackages = getState()?.values?.[input.name] || [];
+    setChosenOptions(sortedOptions(mapPackagesToOptions(loadedPackages)));
+
+    const availableSearchInput = document.querySelector(
+      '[aria-label="available search input"]'
+    );
+
+    availableSearchInput?.addEventListener('keydown', handleSearchOnEnter);
+    return () =>
+      availableSearchInput.removeEventListener('keydown', handleSearchOnEnter);
+  }, []);
+
+  useEffect(() => {
+    if (enterPressed) {
+      handlePackageSearch();
+      setEnterPressed(false);
+    }
+  }, [enterPressed]);
+
+  useEffect(() => {
+    scrollToAddedPackages();
+  }, [scrollTo]);
+
+  const handlePackageSearch = async () => {
+    if (availableInputValue === '') {
+      return;
+    }
+
+    const data = await postRpmsNames(
+      (getState()?.values?.['third-party-repositories'] || [])?.map(
+        ({ URL }) => URL
+      ),
+      availableInputValue
+    );
+
+    if (!data) {
+      setHasNoSearchResults(true);
+      setHasMoreResults(false);
+      setOnlyExactMatch(false);
+      setAvailableOptions([]);
+      return;
+    }
+
+    if (data.length > 100) {
+      setHasNoSearchResults(false);
+      setHasMoreResults(true);
+      setOnlyExactMatch(false);
+
+      let exactMatchIndex = null;
+      data.forEach(({ package_name }, index) => {
+        if (package_name === availableInputValue) {
+          exactMatchIndex = index;
+          return;
+        }
+      });
```

Comment:
```suggestion
      let exactMatchIndex = findExactPackageIndex(data);
```
it could extract into a function that will be easier to read it